### PR TITLE
Allow merging of Namespace and Module in TrieMapping

### LINKED ISSUE #15985: Graph based checking error with CompilationRepresentationFlags.ModuleSuffix
Create a new fsproj, add a.fs, b.fs and c.fs in this order, and enable graph based checking.

a.fs

```fsharp
namespace F.General
```

b.fs

```fsharp
[<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
module F

let br () = ()
```

c.fs

```fsharp
module S

[<EntryPoint>]
let main _ =
    F.br ()
    0
```

Build.

**Expected behavior**

Compiles successfully.

**Actual behavior**

`c.fs(5,5): error FS0039: The value, namespace, type or module 'F' is not defined.` It looks like it has been established that c.fs does not depend on b.fs, and both files are checked in parallel.

**Known workarounds**

Disable graph based checking.

**Related information**

SDK 8.0.100-rc.1.23455.8

### PR DESCRIPTION
**Pull request by nojaf**
Fixes #15985.

@safesparrow would you mind taking a look at this change?
The problem was in constructing the Trie. We didn't anticipate that a name could be used for both a module and a namespace. My fix is to treat the module as it exposes a type inside the namespace.
I think this is acceptable but would like to hear your opinion here.

### CODE REVIEW

**Review by vzarytovskii — APPROVED** [review]

**Comment by vzarytovskii** [comment]
@dotnet/fsharp-team-msft one more review please, @0101 perhaps?

@nojaf do you want to wait for review from Janusz before merge?

**Review by 0101 — APPROVED** [review]
Looks good to me

**Comment by nojaf** [comment]
I spoke with Janusz earlier this week, and he's planning to review it within this week. If you haven't received any updates by next week, I suggest merging it as-is. We can handle any feedback at a later stage.

**Review by safesparrow — APPROVED** [review]
This fixes the issue, so happy for it to be merged. Thanks @nojaf 

Longer term we might want to refactor the model to more accurately represent the language constructs, but that's not a blocker in any way.

**Pull request merged**
